lib/rules/dot-notation.js

```diff
@@ -141,29 +149,24 @@ module.exports = {
                         data: {
                             key: node.property.name
                         },
-                        fix(fixer) {
-                            const dot = sourceCode.getTokenBefore(node.property);
-                            const textAfterDot = sourceCode.text.slice(dot.range[1], node.property.range[0]);
-
-                            if (textAfterDot.trim()) {
-
-                                // Don't perform any fixes if there are comments between the dot and the property name.
-                                return null;
-                            }
+                        *fix(fixer) {
+                            const dotToken = sourceCode.getTokenBefore(node.property);
 
+                            // A statement that starts with `let[` is parsed as a destructuring variable declaration, not a MemberExpression.
                             if (node.object.type === "Identifier" && node.object.name === "let") {
+                                return; // eslint-disable-line eslint-plugin/fixer-return -- false positive
+                            }
```

Comment:
Minor edge case: this might be fine if it's optional `let?.class` -> `let?.["class"]`